server/src/main/java/org/opensearch/action/admin/indices/stats/CommonStatsFlags.java

```diff
@@ -80,6 +83,11 @@ public CommonStatsFlags(StreamInput in) throws IOException {
         if (in.getVersion().onOrAfter(LegacyESVersion.V_7_2_0)) {
             includeUnloadedSegments = in.readBoolean();
         }
+        if (in.getVersion().onOrAfter(LegacyESVersion.V_7_9_0) &&
```

Comment:
OpenSearch users won't have legacy versions >= 7.9. Right?

Reply:
Yeah I had some confusions around it too. Legacy versions for open search will be ``7.10`` as well, however it may not have the shard indexing feature, depending upon the migration type. One options is we only check for ``onOrAfter(Version.V_1_0_0)`` from open search perspective. Alternatively, we keep this check for uniform, across migration experiences, based on the versions >= 7.9 and feature attribute check. Let me know if you recommend keeping ``onOrAfter(Version.V_1_0_0)``  instead.

Reply:
My question is how would a user on Elasticsearch OSS 7.9 will get impacted once they upgrade to Opensearch 1.0.0 with the current logic

Reply:
This change will be kind of no-op for Elasticsearch OSS 7.9/7.10 users during the upgrade flow, and they can start using the feature once on Opensearch 1.x if feature settings are enabled.

Reply:
++